src/box/lua/config/utils/file.lua

```diff
@@ -55,6 +56,77 @@ local function universal_read(file_name, file_kind)
     return data
 end
 
+
+local function get_file_tags(file_name)
+    local function parse_tags(t)
+        local known_tags = {
+            early_load = 'boolean',
+        }
+
+        local tags = {}
+        for k, v in pairs(t) do
+            if known_tags[k] == type(v) then
+                tags[k] = v
+            end
+        end
+
+        return tags
+    end
+
+    -- Tags are only supported for Lua files.
+    if type(file_name) ~= 'string' or not file_name:endswith('.lua') then
+        error(('Invalid file name %q'):format(file_name), 0)
+    end
+
+    local file, err = io.open(file_name, "r")
+    if file == nil then
+        error(('Unable to open %q: %s'):format(file_name, err), 0)
+    end
+
+    local tags = {}
+    local line = file:read("*line")
+
+    -- Skip shebang.
+    if line:startswith('#!') then
+        line = file:read("*line")
+    end
+
+    -- Skip empty lines.
+    while line:lstrip() == '' do
+        line = file:read("*line")
+    end
+
+    -- Trim leading whitespaces.
+    line = string.lstrip(line)
+
+    if line:startswith('--') then
+        local ok, res = pcall(yaml.decode, line:sub(3))
+        if ok and type(res) == 'table' then
+            tags = parse_tags(res)
+        end
+    end
+
+    -- @TODO Add support for multi-line comments.
+
+    file:close()
+    return tags
+end
+
+local function get_module_tags(module_name)
```

Comment:
How about a more general name 'metadata' instead of just 'tags'? We can store arbitrary metadata here, even nested, while 'tags' usually mean a list of associated keywords.

Reply:
Yeah, in the current state the purposed solution became more general. Updated the naming.